File: backend/models/template_model.js
const { DataTypes } = require('sequelize');
const sequelize = require("../config/database");

const Template = sequelize.define('Template', {
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  fileType: {
    type: DataTypes.STRING
  },
  fileUrl: {
    type: DataTypes.STRING,
    allowNull: true,
  },
}, {
  tableName: 'templates',
  timestamps: true
});

Template.sync({alter: false}).then(()=>{
    console.log("Template table created");
}).catch((error)=>{
    console.error("Error creating Template table:", error);
});


module.exports = Template;

File: frontend/lib/models/template_model.dart
class Template {
  final String name;
  final String description;
  final String fileType;
  final String fileUrl;

  Template({
    required this.name,
    required this.description,
    required this.fileType,
    required this.fileUrl,
  });

  factory Template.fromJson(Map<String, dynamic> json) {
    return Template(
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      fileType: json['fileType'] ?? '',
      fileUrl: json['fileUrl'] ?? '',
    );
  }
}
